test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs

```diff
@@ -20,18 +19,22 @@
 
 namespace Microsoft.Health.Dicom.Web.Tests.E2E.Rest;
 
-public class QueryTransactionTests : IClassFixture<HttpIntegrationTestFixture<Startup>>, IAsyncLifetime
+public abstract class QueryTransactionTests : IClassFixture<HttpIntegrationTestFixture<Startup>>, IAsyncLifetime
 {
     private readonly IDicomWebClient _client;
     private readonly DicomInstancesManager _instancesManager;
+    private readonly Action<QueryResource, DicomDataset, DicomDataset> _validateResponseDataset;
 
     public QueryTransactionTests(HttpIntegrationTestFixture<Startup> fixture)
     {
-        EnsureArg.IsNotNull(fixture, nameof(fixture));
-        _client = fixture.GetDicomWebClient();
+        _client = GetClient(fixture);
         _instancesManager = new DicomInstancesManager(_client);
+        _validateResponseDataset = GetValidateResponseDataset();
     }
 
+    protected abstract IDicomWebClient GetClient(HttpIntegrationTestFixture<Startup> fixture);
+    protected abstract Action<QueryResource, DicomDataset, DicomDataset> GetValidateResponseDataset();
```

Comment:
You could simply have abstract method like:

```csharp
protected abstract void ValidateResponseDataset(QueryResource resource, DicomDataset expected, DicomDataset actual);
```

Reply:
Alternatively, instead of using inheritance, you can just make these methods whose test methods are theories and the parameter is the API version